cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
 
-Introduction
-============
+## Context
 
-Content Security Policy (CSP) is an effective "defense in depth" technique to be used against content injection attacks. It is a declarative policy that informs the user agent what are valid sources to load from.
+The increase in XSS and clickjacking vulnerabilities demands a more `defense in depth` security approach. CSP comes in place to enforce the loading of resources (scripts, images, etc.) from restricted locations that are trusted by the server, as well as enforcing HTTPS usage transparently. Moreover, the developer will get more visibility on the attacks occurring on the application by using the CSP reporting directive.
 
-Since, it was introduced in Firefox version 4 by Mozilla, it has been adopted as a standard, and grown in adoption and capabilities.
+## Avoid CSP
 
-This document is meant to provide guidance on how to utilize CSP under a variety of situations to address a variety of concerns.
+If you are a developer of any of the applications mentioned below, CSP will barely provide or improve their security:
+- Single page applications with no cookies or authentication.
+- Applications that are already vulnerable to XSS vulnerabilities and chose not to remediate them. CSP is not the first line of defense.
 
-References
-==========
+# Policy Delivery
 
-Specifications of the CSP standard can be found the following locations:
+CSP can be delivered to the user agent in different techniques.
+1. `Content-Security-Policy` HTTP response header field. This is the most preferred technique.
+1. `<meta>` HTML element with `http-equiv` attribute set to `Content-Security-Policy`. These elements need to be placed as early as possible in the documents.
+1. `Content-Security-Policy-Report-Only` HTTP response header field. This header is used when the developer is unsure of the CSP behavior and wants to monitor it, instead of enforcing it.
 
--   Latest Revision - <https://w3c.github.io/webappsec/specs/content-security-policy/>
--   Latest Version (CSP2) - <http://www.w3.org/TR/CSP2/>
--   CSP 1.0 - <http://www.w3.org/TR/2012/CR-CSP-20121115/>
+## HTTP Headers
 
-CSP Basics
-==========
-
-CSP consists of a series of directives. CSP has also evolved over two major revisions. Most browsers support 1.0, and adoption of CSP2 has been incremental.
+The following are headers for CSP.
 
-HTTP Headers
-------------
+- `Content-Security-Policy` : W3C Spec standard header. Supported by Firefox 23+, Chrome 25+ and Opera 19+
+- `Content-Security-Policy-Report-Only` : W3C Spec standard header. Supported by Firefox 23+, Chrome 25+ and Opera 19+, whereby the policy is non-blocking ("fail open") and a report is sent to the URL designated by the `report-uri` directive. This is often used as a precursor to utilizing CSP in blocking mode ("fail closed")
```

Comment:
Can be interesting to mention (perhaps it is done in the rest of the CS but i add the note here) that injection can be performed in a CSP notificatin message so the endpoint in charge of handling of CSP notification must perform input validation using [JSON Schema](https://json-schema.org/) for example or a service like [report-uri](https://report-uri.com/) can be used.